### PR TITLE
JAMES-1839 SimpleGroupMembershipResolver should not fail badly on nul…

### PR DESCRIPTION
**Pull request by Laura-Royet**
…l user

### CODE REVIEW

File: mailbox/api/src/test/java/org/apache/james/mailbox/acl/SimpleGroupMembershipResolverTest.java [hunk 1]
```diff
@@ -0,0 +1,42 @@
+package org.apache.james.mailbox.acl;
```
Comment (chibenwa) [inline mailbox/api/src/test/java/org/apache/james/mailbox/acl/SimpleGroupMembershipResolverTest.java]: License

File: mailbox/api/src/test/java/org/apache/james/mailbox/acl/SimpleGroupMembershipResolverTest.java [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+    SimpleGroupMembershipResolver simpleGroupMembershipResolver;
+
+    @Before
+    public void setUp() {
+       simpleGroupMembershipResolver = new SimpleGroupMembershipResolver();
+    }
+
+    @Test
+    public void isMemberShoulReturnFalseWithAnEmptySimpleGroupMembershipResolver() throws Exception {
```
Comment (chibenwa) [inline mailbox/api/src/test/java/org/apache/james/mailbox/acl/SimpleGroupMembershipResolverTest.java]: Should

File: mailbox/api/src/test/java/org/apache/james/mailbox/acl/SimpleGroupMembershipResolverTest.java (hunk 1 above)
Comment (Laura-Royet) [inline mailbox/api/src/test/java/org/apache/james/mailbox/acl/SimpleGroupMembershipResolverTest.java]: ok, added.

File: mailbox/api/src/test/java/org/apache/james/mailbox/acl/SimpleGroupMembershipResolverTest.java (hunk 2 above)
Comment (Laura-Royet) [inline mailbox/api/src/test/java/org/apache/james/mailbox/acl/SimpleGroupMembershipResolverTest.java]: ok, corrected

**Comment by chibenwa** [comment]
test this please

**Comment by Laura-Royet** [comment]
test this please

File: mailbox/api/src/test/java/org/apache/james/mailbox/acl/SimpleGroupMembershipResolverTest.java [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+package org.apache.james.mailbox.acl;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import org.junit.Before;
+import org.junit.Test;
+
+public class SimpleGroupMembershipResolverTest {
+
+    SimpleGroupMembershipResolver simpleGroupMembershipResolver;
```
Comment (aduprat) [inline mailbox/api/src/test/java/org/apache/james/mailbox/acl/SimpleGroupMembershipResolverTest.java]: private

File: mailbox/api/src/test/java/org/apache/james/mailbox/acl/SimpleGroupMembershipResolverTest.java [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+
+    @Before
+    public void setUp() {
+       simpleGroupMembershipResolver = new SimpleGroupMembershipResolver();
+    }
+
+    @Test
+    public void isMemberShouldReturnFalseWithAnEmptySimpleGroupMembershipResolver() throws Exception {
```
Comment (aduprat) [inline mailbox/api/src/test/java/org/apache/james/mailbox/acl/SimpleGroupMembershipResolverTest.java]: s/With/When
also for other methods

File: mailbox/api/src/test/java/org/apache/james/mailbox/acl/SimpleGroupMembershipResolverTest.java [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+        //Then
+        assertThat(testee).isFalse();
+    }
+
+    @Test
+    public void addMembershipShouldAddAMembershipWithANonNullUser() throws Exception {
+        //When
+        simpleGroupMembershipResolver.addMembership("group", "user");
+        //Then
+        assertThat(simpleGroupMembershipResolver.isMember("user", "group")).isTrue();
```
Comment (aduprat) [inline mailbox/api/src/test/java/org/apache/james/mailbox/acl/SimpleGroupMembershipResolverTest.java]: extract the return value of the tested method.

File: mailbox/api/src/test/java/org/apache/james/mailbox/acl/SimpleGroupMembershipResolverTest.java (hunk 3 above)
Comment (Laura-Royet) [inline mailbox/api/src/test/java/org/apache/james/mailbox/acl/SimpleGroupMembershipResolverTest.java]: Ok, done

File: mailbox/api/src/test/java/org/apache/james/mailbox/acl/SimpleGroupMembershipResolverTest.java (hunk 4 above)
Comment (Laura-Royet) [inline mailbox/api/src/test/java/org/apache/james/mailbox/acl/SimpleGroupMembershipResolverTest.java]: ok, done

File: mailbox/api/src/test/java/org/apache/james/mailbox/acl/SimpleGroupMembershipResolverTest.java (hunk 5 above)
Comment (Laura-Royet) [inline mailbox/api/src/test/java/org/apache/james/mailbox/acl/SimpleGroupMembershipResolverTest.java]: Done

File: mailbox/api/src/test/java/org/apache/james/mailbox/acl/SimpleGroupMembershipResolverTest.java [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+
+    @Before
+    public void setUp() {
+       simpleGroupMembershipResolver = new SimpleGroupMembershipResolver();
+    }
+
+    @Test
+    public void isMemberShouldReturnFalseWhenEmptySimpleGroupMembershipResolver() throws Exception {
+        //When
+        boolean testee = simpleGroupMembershipResolver.isMember("user", "group");
```
Comment (mbaechler) [inline mailbox/api/src/test/java/org/apache/james/mailbox/acl/SimpleGroupMembershipResolverTest.java]: it's not testee

File: mailbox/api/src/test/java/org/apache/james/mailbox/acl/SimpleGroupMembershipResolverTest.java [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+
+    private SimpleGroupMembershipResolver simpleGroupMembershipResolver;
+
+    @Before
+    public void setUp() {
+       simpleGroupMembershipResolver = new SimpleGroupMembershipResolver();
+    }
+
+    @Test
+    public void isMemberShouldReturnFalseWhenEmptySimpleGroupMembershipResolver() throws Exception {
```
Comment (mbaechler) [inline mailbox/api/src/test/java/org/apache/james/mailbox/acl/SimpleGroupMembershipResolverTest.java]: you can make the method name shorter : isMemberShouldReturnFalseWhenEmptyResolver

File: mailbox/api/src/test/java/org/apache/james/mailbox/acl/SimpleGroupMembershipResolverTest.java [hunk 8]
```diff
@@ -0,0 +1,63 @@
...
+
+    @Test
+    public void addMembershipShouldAddAMembershipWithANullUser() throws Exception {
+        //When
+        simpleGroupMembershipResolver.addMembership("group", null);
+        boolean actual = simpleGroupMembershipResolver.isMember(null, "group");
+        //Then
+        assertThat(actual).isTrue();
+    }
+
```
Comment (mbaechler) [inline mailbox/api/src/test/java/org/apache/james/mailbox/acl/SimpleGroupMembershipResolverTest.java]: you lack test with isMemberReturning true

File: mailbox/api/src/test/java/org/apache/james/mailbox/acl/SimpleGroupMembershipResolverTest.java [hunk 9]
```diff
@@ -0,0 +1,63 @@
...
+        simpleGroupMembershipResolver.addMembership("group", "user");
+        boolean actual = simpleGroupMembershipResolver.isMember("user", "group");
+        //Then
+        assertThat(actual).isTrue();
+    }
+
+    @Test
+    public void addMembershipShouldAddAMembershipWithANullUser() throws Exception {
+        //When
+        simpleGroupMembershipResolver.addMembership("group", null);
```
Comment (mbaechler) [inline mailbox/api/src/test/java/org/apache/james/mailbox/acl/SimpleGroupMembershipResolverTest.java]: give a name to your null values

File: mailbox/api/src/test/java/org/apache/james/mailbox/acl/SimpleGroupMembershipResolverTest.java (hunk 6 above)
Comment (Laura-Royet) [inline mailbox/api/src/test/java/org/apache/james/mailbox/acl/SimpleGroupMembershipResolverTest.java]: ok, replaced by "actual"

File: mailbox/api/src/test/java/org/apache/james/mailbox/acl/SimpleGroupMembershipResolverTest.java (hunk 9 above)
Comment (Laura-Royet) [inline mailbox/api/src/test/java/org/apache/james/mailbox/acl/SimpleGroupMembershipResolverTest.java]: Done

File: mailbox/api/src/test/java/org/apache/james/mailbox/acl/SimpleGroupMembershipResolverTest.java (hunk 8 above)
Comment (Laura-Royet) [inline mailbox/api/src/test/java/org/apache/james/mailbox/acl/SimpleGroupMembershipResolverTest.java]: Done

**Comment by Laura-Royet** [comment]
test this please

**Pull request merged**